fenwick/src/main/java/org/opencadc/fenwick/ArtifactSync.java

```diff
@@ -0,0 +1,184 @@
+                        final Date currLastModified, final StorageSite storageSite) {
+        this.selector = selector;
+        this.tapClient = tapClient;
+        this.currLastModified = currLastModified;
+        this.storageSite = storageSite;
+    }
+
+    /**
+     * Execute the query and return the iterator back.
+     *
+     * @return ResourceIterator over Artifact instances matching the selector's query parameters.
+     *
+     * @throws ResourceNotFoundException For any missing required configuration that is missing.
+     * @throws IOException               For unreadable configuration files.
+     * @throws IllegalStateException     For any invalid configuration.
+     * @throws TransientException        temporary failure of TAP service: same call could work in future
+     * @throws InterruptedException      thread interrupted
+     */
+    public ResourceIterator<Artifact> iterator()
+            throws ResourceNotFoundException, IOException, IllegalStateException, TransientException,
+                   InterruptedException {
+
+        final String query = assembleQuery();
+        LOGGER.debug("\nExecuting query '" + query + "'\n");
+        return tapClient.execute(query, row -> {
+            int index = 0;
+            final Artifact artifact = new Artifact((UUID) row.get(index++),
+                                                   (URI) row.get(index++),
+                                                   (URI) row.get(index++),
+                                                   (Date) row.get(index++),
+                                                   (Long) row.get(index++));
+            artifact.contentType = row.get(index++).toString();
+            artifact.contentEncoding = row.get(index++).toString();
+
+            InventoryUtil.assignMetaChecksum(artifact, (URI) row.get(index++));
+            InventoryUtil.assignLastModified(artifact, (Date) row.get(index));
+
+            artifact.siteLocations.add(new SiteLocation(storageSite.getID()));
+
+            return artifact;
+        });
+    }
+
+    /**
+     * Create a query for the TAP (Luskan) query.  This method will tally up the selector's clauses into a formulated
+     * query for the TapClient.
+     *
+     * <p>A potential risk with this method is that there is no enforcement for at least one clause.  This could result
+     * in a very large query.
+     *
+     * @return String ADQL query with expected WHERE clauses.
+     *
+     * @throws ResourceNotFoundException For any missing required configuration that is missing.
+     * @throws IOException               For unreadable configuration files.
+     * @throws IllegalStateException     For any invalid configuration.
+     */
+    private String assembleQuery() throws ResourceNotFoundException, IOException, IllegalStateException {
+        final StringBuilder whereClauseBuilder = new StringBuilder();
+        final Iterator<String> whereClauseIterator = selector.iterator();
+
```

Comment:
Each string in the iterator is from a different include file so defined a different filter... they cannot be added together with AND becaue they most likely select different sets of artifacts. In the query here, you need to have only one of those.

and that whole clause has to be  in `( )` in case it has `OR` in it